File: 2-introduction-to-deep-agents/typescript/m2/m2.4_eval_agent.ts
// typescript/m2/m2.4_eval_agent.ts
import { createCodeInterpreterMiddleware } from "@langchain/quickjs";
import { createDeepAgent } from "deepagents";

import { model } from "../models.js";

const agent = createDeepAgent({
  model,
  middleware: [createCodeInterpreterMiddleware()],
});

const result = await agent.invoke({
  messages: [
    {
      role: "user",
      content: "Use the eval tool to compute and return the first 15 Fibonacci numbers.",
    },
  ],
});

console.log(result.messages.at(-1)?.content);

File: 2-introduction-to-deep-agents/python/m2/m2.4_eval_agent.py
from deepagents import create_deep_agent
from langchain_quickjs import CodeInterpreterMiddleware

from models import model

agent = create_deep_agent(
    model=model,
    middleware=[CodeInterpreterMiddleware()],
)

result = agent.invoke(
    {
        "messages": [
            {
                "role": "user",
                "content": "Use the eval tool to compute and return the squares of the numbers 1 to 20 inclusive.",
            }
        ]
    }
)

print(result["messages"][-1].content)
